Content.Trauma.Client/Botany/PlantAnalyzer/UI/PlantAnalyzerWindow.xaml.cs

```diff
@@ -0,0 +1,296 @@
+    {
+        _internalGeneDatabank = msg.SeedData;
+        _internalConsumeGasesDatabank = msg.ConsumeGasData;
+        _internalExudeGasesDatabank = msg.ExudeGasData;
+        _internalChemicalsDatabank = msg.ChemicalData;
+    }
+
+    public void PopulateUpdateButtons()
+    {
+        switch (_internalmode)
+        {
+            case PlantAnalyzerModes.Scan:
+                Tabs.CurrentTab = 0;
+                break;
+            case PlantAnalyzerModes.DeleteMutations:
+                Tabs.CurrentTab = 1;
+                break;
+            case PlantAnalyzerModes.Extract:
+                Tabs.CurrentTab = 2;
+                if (_internalGeneNumber < 0)
+                {
+                    GeneIndexExtract.Text = Loc.GetString($"plant-analyzer-invalid-gene");
+                    return;
+                }
+                GeneIndexExtract.Text = SeedDataTypes.IdToString[_internalGeneNumber];
+                break;
+            case PlantAnalyzerModes.Implant:
+                Tabs.CurrentTab = 3;
+                if (_internalGeneDatabank.Count + _internalConsumeGasesDatabank.Count + _internalExudeGasesDatabank.Count + _internalChemicalsDatabank.Count <= 0)
+                {
+                    GeneIndexDatabase.Text = Loc.GetString($"plant-analyzer-invalid-database");
+                }
+                else
+                {
+                    GeneIndexDatabase.Text = "Selected ID: " + (_internalDatabaseNumber + 1).ToString();
+                }
+                GeneDatabaseListContainer.RemoveAllChildren();
+                GeneDatabaseListContainer.AddChild(new Label
+                {
+                    Text = Loc.GetString("plant-analyzer-database-entries-header"),
+                    Margin = new Thickness(10, 0, 0, 0),
+                });
+                int index = 0;
+                foreach (GeneData gene in _internalGeneDatabank)
+                {
+                    int mutationID = gene.GeneID;
+                    SeedDataTypes.SeedDataType mutationType = SeedDataTypes.IdToType[mutationID];
+                    string mutationValue = mutationType switch
+                    {
+                        SeedDataTypes.SeedDataType.Float => $"{gene.GeneValue:F2}",
+                        SeedDataTypes.SeedDataType.Int => $"{(int) gene.GeneValue:D0}",
+                        SeedDataTypes.SeedDataType.HarvestType => Loc.GetString($"plant-analyzer-harvest-{(AnalyzerHarvestType) (int) gene.GeneValue}"),
+                        SeedDataTypes.SeedDataType.Bool => gene.GeneValue >= 0.5f ? Loc.GetString("plant-analyzer-boolean-true") : Loc.GetString("plant-analyzer-boolean-false"),
+                        _ => "N/A"
+                    };
+                    GeneDatabaseListContainer.AddChild(new RichTextLabel
+                    {
+                        Text = "ID: " + (index + 1) + " " + SeedDataTypes.IdToString[mutationID] + ": " + mutationValue,
```

Comment:
use `$"{whatever} {foo}: {bar}" for all of these instead of adding 5 strings

store `"ID: " + (index + 1)` in a variable and reuse it